editor/uid_upgrade_tool.cpp

```diff
@@ -0,0 +1,126 @@
+}
+
+void UIDUpgradeTool::begin_upgrade() {
+	EditorSettings::get_singleton()->set_project_metadata("uid_upgrade_tool", "run_on_restart", false);
+}
+
+void UIDUpgradeTool::finish_upgrade() {
+	EditorNode::get_singleton()->trigger_menu_option(EditorSceneTabs::SCENE_CLOSE_ALL, true);
+
+	Vector<String> resave_paths = EditorSettings::get_singleton()->get_project_metadata("uid_upgrade_tool", "resave_paths", Variant());
+	EditorProgress ep("uid_upgrade_resave", TTR("Updating Script UIDs"), resave_paths.size());
+
+	int step = 0;
+	for (const String &file_path : resave_paths) {
+		Ref<Resource> res = ResourceLoader::load(file_path);
+		ep.step(TTR("Attempting to re-save ") + file_path, step++);
+		if (res.is_valid()) {
+			ResourceSaver::save(res);
+		}
+	}
+	EditorSettings::get_singleton()->set_project_metadata("uid_upgrade_tool", "resave_paths", Vector<String>());
+
+	emit_signal(SNAME("upgrade_finished"));
+}
+
+void UIDUpgradeTool::_bind_methods() {
+	ADD_SIGNAL(MethodInfo("upgrade_finished"));
+}
+
+UIDUpgradeTool::UIDUpgradeTool() {
+	singleton = this;
+}
+
+UIDUpgradeTool::~UIDUpgradeTool() {
+	singleton = nullptr;
+}
+
+void UIDUpgradeDialog::popup_on_demand() {
+	const String confirmation_message = TTR("As of Godot 4.4, scenes and resources use UIDs to reference scripts and shaders.\n\nNormally, this update is applied to a single scene or resource once you save it in Godot 4.4 for the first time. If you have a lot of scenes and/or resources, doing this manually may be time-consuming. This tool will update all of the project's scenes and resources at once.\n\nClick \"Restart & Upgrade\" to restart the editor and update all of the scenes and resources in this project. Depending on the project size, it may take several minutes.\n\nNote: Please make sure you have your project backed up before running the tool, to avoid the possibility of data loss. Additionally, make sure to commit all .uid files into version control (and do not add them to ignore-lists like .gitignore).");
+	set_text(confirmation_message);
+	get_ok_button()->set_text("Restart & Upgrade");
+	popup_centered(Size2(750 * EDSCALE, 0));
+}
+
+void UIDUpgradeDialog::_on_custom_action(const String &p_action) {
+	if (p_action == "uid_upgrade_learn_more") {
+		OS::get_singleton()->shell_open("https://godotengine.org/article/uid-changes-coming-to-godot-4-4/");
+	}
+}
+
+void UIDUpgradeDialog::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_READY:
+			connect(SceneStringName(confirmed), callable_mp(UIDUpgradeTool::get_singleton(), &UIDUpgradeTool::prepare_upgrade));
+			connect(SNAME("custom_action"), callable_mp(this, &UIDUpgradeDialog::_on_custom_action));
+			break;
+	}
+}
+
+UIDUpgradeDialog::UIDUpgradeDialog() {
+	set_autowrap(true);
+	get_label()->set_custom_minimum_size(Size2(750 * EDSCALE, 0));
+	learn_more_button = add_button(TTR("Learn more"), true, "uid_upgrade_learn_more");
```

Comment:
Use Title Case for buttons:

```suggestion
	learn_more_button = add_button(TTR("Learn More"), true, "uid_upgrade_learn_more");
```